mu-plugins/blocks/multisite-latest-posts/multisite-latest-posts.php

```diff
@@ -0,0 +1,165 @@
+<?php
+/**
+ * Plugin Name: Multisite Latest Posts
+ * Description: A block for use across the whole wp.org network.
+ *
+ * @package wporg
+ */
+
+namespace WordPressdotorg\MU_Plugins\wporg;
+
+/**
+ * Helper function to find item in array by id.
+ *
+ * @param WP_Term[] $arr List of categories
+ * @param number    $id Category ID
+ * @return WP_Term|string
+ */
+function get_by_id( $arr, $id ) {
+	foreach ( $arr as $item ) {
+		if ( $id == $item->id ) {
+			return $item;
+		}
+	}
+
+	return '';
+}
+
+/**
+ * Fetches categories from REST endpoint.
+ *
+ * @param string $endpoint URL
+ * @return WP_Term[]|WP_Error
+ */
+function get_categories_via_api( $endpoint ) {
+	$response = wp_remote_get( esc_url_raw( $endpoint . '/categories' . '?per_page=100' ) );
+
+	if ( is_wp_error( $response ) ) {
+		return new \WP_Error( 500, __( 'An error has occurred fetching categories.', 'wporg' ) );
+	}
+
+	$body = wp_remote_retrieve_body( $response );
+
+	if ( empty( $body ) ) {
+		return new \WP_Error( 500, __( 'An error has occurred fetching categories.', 'wporg' ) );
+	}
+
+	return json_decode( $body );
+}
+
+/**
+ * Returns the category based on its ID.
+ *
+ * @param string $endpoint URL
+ * @param number $id Category ID
+ * @return WP_Term[]|string
+ */
+function get_category( $endpoint, $id ) {
+	$categories = get_categories_via_api( $endpoint );
+
+	// We are okay if we can't find the category
+	if ( is_wp_error( $categories ) ) {
+		return '';
+	}
+
+	return get_by_id( $categories, $id );
+}
+
+/**
+ * Returns a list of posts.
+ *
+ * @param string  $endpoint URL
+ * @param string  $post_type WP_Post_Type label
+ * @param integer $limit Numbers of posts to return
+ * @return WP_Post[]|WP_Error
+ */
+function get_posts_via_api( $endpoint, $post_type = 'posts', $limit = 10 ) {
```

Comment:
I note that none of these api methods have any form of caching - less important for `switch_to_blog()` but highly important for HTTP calls.

I'd probably suggest that `get_posts_via_api()` use a transient to cache the data.
`
$data = get_transient( __NAMESPACE__ );
if ( $data ) {
   return $data;
}
`
and
`set_transient( __NAMESPACE__, $data, HOUR_IN_SECONDS );` before returning at the end.

Ideally, `get_category()` would also be called in here prior to the data being cached, so it's just a singular transient that needs clearing.

Reply:
I always forget the cache. Thanks. 

It would be nice if we had middleware to do the caching.

Reply:
Do you think we'll land a fix for the news podcast embed issue? If so I'll leave out category caching for now because I will remove it.

Reply:
> Do you think we'll land a fix for the news podcast embed issue?

Tomorrow hopefully

Reply:
@StevenDufresne FYI; The podcast category embed issue should now be resolved, in a way that won't require reverts again in future..